packages/client/src/publish/Publisher.ts

```diff
@@ -24,13 +27,21 @@ const parseTimestamp = (metadata?: MessageMetadata): number => {
 export class Publisher implements Context {
     readonly id
     readonly debug
+    getGroupKeyQueue: (streamId: StreamID) => Promise<GroupKeyQueue>
 
     constructor(
         context: Context,
-        @inject(delay(() => PublishPipeline)) private pipeline: PublishPipeline
+        @inject(delay(() => PublishPipeline)) private pipeline: PublishPipeline,
+        groupKeyStoreFactory: GroupKeyStoreFactory
     ) {
         this.id = instanceId(this)
         this.debug = context.debug.extend(this.id)
+        this.getGroupKeyQueue = pMemoize(async (streamId: StreamID) => {
```

Comment:
Why memoized?

Reply:
For each `streamId` we should keep one group key queue. This way we get the same instance for each call.